### PR TITLE
Use header instead of extern for bam_cat

### PR DESCRIPTION
**Pull request by alexf101**
Include bam_cat.c since it's used in STAR.cpp.

Delete commented-out includes while we're here.

### CODE REVIEW

**Comment by alexdobin** [comment]
Hi Alex,

I have bam_cat defined as extern in STAR.cpp -  is this not working for some reason?
Then I would rather make a bam_cat.h header, so that the code compiles separately from STAR.

Cheers
Alex

**Comment by alexf101** [comment]
Hi Alex!

Yeah, having bam_cat as an extern was making trouble in my build environment - not 100% sure why.

I agree with you that the neatest option is to create a header for bam_cat, so I've altered this patch to do that instead.

Cheers,

Alex

File: source/sjSplitAlign.cpp [hunk 1]
```diff
@@ -1,3 +1,6 @@
+#include "IncludeDefine.h"
```
Comment (alexf101) [inline source/sjSplitAlign.cpp]: Not sure why this is showing up - I thought this had already been merged upstream? Perhaps git is just having a moment :)

Comment (alexdobin) [inline source/sjSplitAlign.cpp]: It worked for me... the line appears in my local file.

**Comment by alexdobin** [comment]
Merged!

**Pull request merged**